bin/run-browser-tests-ci-batch

```diff
@@ -4,43 +4,61 @@
 # DOC: 1-indexed batch number.
 
 import glob
-import math
 import subprocess
 import sys
 
 BATCH_COUNT = 10
 
 
-def bad_batch_number_input(batch_num_input):
+def handle_bad_input(batch_num_input):
     print(f"Invalid batch number: {batch_num_input}", file=sys.stderr)
     exit(1)
 
 
-if len(sys.argv) < 2:
-    bad_batch_number_input("none")
+# Gets the batch number from the command line arguments. The batch number is *one-indexed*.
+def get_batch_number():
+    if len(sys.argv) < 2:
+        handle_bad_input("none")
+    try:
+        batch_number = int(sys.argv[1])
+    except ValueError:
+        handle_bad_input(sys.argv[1])
+    if batch_number < 1 or batch_number > BATCH_COUNT:
+        handle_bad_input(batch_number)
+    return batch_number
 
-try:
-    batch_number = int(sys.argv[1])
-except ValueError:
-    bad_batch_number_input(sys.argv[1])
 
-if batch_number < 1 or batch_number > BATCH_COUNT:
-    bad_batch_number_input(batch_number)
+# Gets a list of all the test files in the browser-test directory.
+def get_all_test_files():
+    test_files = glob.glob('browser-test/src/*.test.ts')
+    test_files = [file.replace("browser-test/", "") for file in test_files]
+    test_files.sort()
+    return test_files
 
-test_files = glob.glob('browser-test/src/*.test.ts')
-test_files = [file.replace("browser-test/", "") for file in test_files]
-test_files.sort()
-batch_size = math.ceil(len(test_files) / BATCH_COUNT)
 
-batch_contents = test_files[(batch_number - 1) * batch_size:batch_number *
-                            batch_size]
+# Returns a subset of the test files, based on the batch number. This ensures we
+# are evenly distributing the test across all available GitHub Actions runners.
+def get_batch():
+    all_test_files = get_all_test_files()
+    batch_files = []
```

Comment:
**suggestion:** If I'm remembering my python correctly, I think you could do something like `all_test_files[get_batch_number()-1::BATCH_COUNT]` to get every nth index.

Reply:
Nice catch!